Add missing wavelengthrange files

rosteen opened the pull request:
Added file with all wavelength ranges for HST and JWST to `common` config folder, generalizing the wavelength range creation function to generate asdf files for all HST grisms from that file.

duytnguyendtn commented on astrogrism/config/common/WFSS_Wavelength_Range_V1.1_with_ACS.tab:
```diff
@@ -0,0 +1,178 @@
+JWST NIRISS GR150C "" F150W +1 1.288 1.712
+JWST NIRISS GR150C "" F150W 0 1.288 1.712
+JWST NIRISS GR150C "" F150W +2 1.3387046941544807 1.7143780419540788
+JWST NIRISS GR150C "" F150W +3 1.3493257021002125 1.7143780419540788
+JWST NIRISS GR150C "" F150W -1 1.3493257021002125 3.0510001953125
+JWST NIRISS GR150R "" F090W +1 0.7887001774306214 1.031
+JWST NIRISS GR150R "" F090W 0 0.7887001774306214 1.031
+JWST NIRISS GR150R "" F090W +2 0.7865217488098764 1.0265246003059216
+JWST NIRISS GR150R "" F090W +3 0.787 1.029
+JWST NIRISS GR150R "" F090W -1 0.787454623338691 3.396
+JWST NIRISS GR150R "" F200W +1 1.7029009582509522 2.279451922495031
+JWST NIRISS GR150R "" F200W 0 1.7029009582509522 2.279451922495031
+JWST NIRISS GR150R "" F200W +2 1.7096428456142785 2.280352767465269
+JWST NIRISS GR150R "" F200W +3 1.7103185006448478 2.280352767465269
+JWST NIRISS GR150R "" F200W -1 1.713 2.2810001953125
+JWST NIRISS GR150R "" F115W +1 0.9973383916486066 1.3187545575193975
+JWST NIRISS GR150R "" F115W 0 0.9973383916486066 1.3187545575193975
+JWST NIRISS GR150R "" F115W +2 0.9926207124409168 1.2919352743310675
+JWST NIRISS GR150R "" F115W +3 0.993 1.2934676076862788
+JWST NIRISS GR150R "" F115W -1 0.9937980359933947 3.092
+JWST NIRISS GR150R "" F150W +1 1.288 1.712
+JWST NIRISS GR150R "" F150W 0 1.288 1.712
+JWST NIRISS GR150R "" F150W +2 1.3387046941544807 1.7143780419540788
+JWST NIRISS GR150R "" F150W +3 1.3493257021002125 1.7143780419540788
+JWST NIRISS GR150R "" F150W -1 1.3493257021002125 3.0510001953125
+JWST NIRISS GR150C "" F090W +1 0.7887001774306214 1.031
+JWST NIRISS GR150C "" F090W 0 0.7887001774306214 1.031
+JWST NIRISS GR150C "" F090W +2 0.7865217488098764 1.0265246003059216
+JWST NIRISS GR150C "" F090W +3 0.787 1.029
+JWST NIRISS GR150C "" F090W -1 0.787454623338691 3.396
+JWST NIRISS GR150C "" F200W +1 1.7029009582509522 2.279451922495031
+JWST NIRISS GR150C "" F200W 0 1.7029009582509522 2.279451922495031
+JWST NIRISS GR150C "" F200W +2 1.7096428456142785 2.280352767465269
+JWST NIRISS GR150C "" F200W +3 1.7103185006448478 2.280352767465269
+JWST NIRISS GR150C "" F200W -1 1.713 2.2810001953125
+JWST NIRISS GR150C "" F115W +1 0.9973383916486066 1.3187545575193975
+JWST NIRISS GR150C "" F115W 0 0.9973383916486066 1.3187545575193975
+JWST NIRISS GR150C "" F115W +2 0.9926207124409168 1.2919352743310675
+JWST NIRISS GR150C "" F115W +3 0.993 1.2934676076862788
+JWST NIRISS GR150C "" F115W -1 0.9937980359933947 3.092
```
I'm all for eventually expanding astrogrism beyond HST, and I think including these values puts us in a good spot to expand our work. I do wonder whether we should have them here though if we don't do anything with JWST yet?

rosteen replied:
I figured it's better to have them then remove them and have to track them down again later. Doesn't hurt anything to have them here, this isn't a file users should ever be looking at 🤷

duytnguyendtn commented:
Any chance you'll get the chance to fix the codestyle issues to get the tests to run?

rosteen commented:
Whoops, hadn't noticed that there were style errors. One sec...

duytnguyendtn commented:
Sanity check: Are we even using the `wavelengthrange` reference file? I was actually surprised to see you rename the reference files, and have the tests magically pass. Then I looked into the package and I only see the `distortion` and `specwcs` files being called: https://github.com/spacetelescope/astrogrism/blob/main/astrogrism/grism_observation.py#L148-L151

Please tell me I'm crazy 😅

rosteen commented:
I don't think we use them anywhere currently, I was hoping they would be useful in the future (e.g. to limit the wavelength ranges when extracting or simulating spectra).

duytnguyendtn commented:
Interesting! I recall were necessary to construct the nircam imaging pipeline in the sandbox; I actually wasn't aware we didn't need them for our `GrismObs`

I guess in that case, there isn't too much to check here. Thanks for fixing the codestyle; I'll go ahead and merge!

duytnguyendtn reviewed (commented):
In the original wavelengthrange file, the wavelengthranges are bundled in the `wavelengthrange` section:
https://github.com/spacetelescope/astrogrism/pull/21/files#diff-81eef8834b09f29af4e47a7bfe655def98685a740daa40feebb6b49af77a2347L43-L54

In the new wavelengthrange files, I actually don't see the `wavelengthrange` section at all. I actually don't even see the values at all in the reference files. Am I missing something?

rosteen commented:
I actually think I'm slightly wrong, reading https://jwst-pipeline.readthedocs.io/en/latest/jwst/references_general/wavelengthrange_reffile.html it seems like our format is right for the NIRCAM and NIRISS schemas (and a couple others) but it's now getting validated against the NIRSpec schema for some reason. Probably some variable I changed because I didn't think the "TS" in "TSGRISM" was relevant to HST stuff.

rosteen commented:
@duytnguyendtn Fixed, turns out I just needed to convert some strings from the csv read to floats/ints where appropriate.

rosteen commented:
@duytnguyendtn I think I resolved the file move successfully, could you confirm that it looks good?

duytnguyendtn commented:
Roger @rosteen! Seeing failing codechecks again, but I'll start reviewing in the meantime

rosteen commented:
Whoops, fixed.

duytnguyendtn reviewed (changes requested):
When I install this PR fresh, it looks like the wavelength range table file is missing:

```
FileNotFoundError: [Errno 2] No such file or directory: '...\\envpr21\\lib\\site-packages\\astrogrism\\config\\common\\WFSS_Wavelength_Range_V1.1_with_ACS.tab'
```

Quick google search shows that our entry in the `setup.cfg` 

https://github.com/spacetelescope/astrogrism/blob/63ce2613067fe8e17f5a605c411a5d27b7d16634/setup.cfg#L48-L49

doesn't load files recursively. Found a thread with a few ideas for you to peruse: https://stackoverflow.com/questions/27664504/how-to-add-package-data-recursively-in-python-setup-py

rosteen commented:
> doesn't load files recursively

Not a recursion issue, it's another folder at the same level. I added the new folder to the `package_data` and fixed the other outdated function name you found. Good catches!

duytnguyendtn commented on astrogrism/HST/reference_file_generators/generate_wavelengthrange.py:
```diff
@@ -13,12 +15,11 @@
-                                   extract_orders=None):
-    """Create a wavelengthrange reference file for WFC3 TSGRISM mode.
+def create_grism_wavelengthrange(grism, chip=None,
+                                 outname="wfc3_grism_wavelengthrange.asdf",
```
In the similar direciton as your other changes, I think it's appropriate now that we have the other instruments to change the `wfc3` default to just `hst` default? Or maybe even just drop it?

``` suggestion
                                 outname="hst_grism_wavelengthrange.asdf",
```

rosteen replied:
Good call. Rather than having a default it's probably better to just generate the right name based on the grism selected... I can code that in, give me a minute...

duytnguyendtn reviewed (approved):
One small last nitpick, but was able to run the generate reffiles notebook. OK. Thanks for the improvement and changes! One small detail I noticed was the file naming in the notebook just assigns the name of the file to something generic `astropy_downloadfile..._wavelengthfile.asdf`. We might want to later improve this to something more intelligent: https://github.com/spacetelescope/astrogrism/blob/679bfca07168923e928fa796cd1402a4661a6c93/astrogrism/HST/create_HST_reffiles.py#L30

rosteen commented:
@duytnguyendtn think that's better?

rosteen commented:
It looked like internet connection problems, not code problems.

duytnguyendtn commented:
Looks like it failed to connect to PyPI. I kicked the tests to run again
